presentation/src/main/java/com/sopt/presentation/ui/screen/signin/viewmodel/SignInViewModel.kt

```diff
@@ -27,53 +23,87 @@ class SignInViewModel @Inject constructor(
+                    is SignInError.PasswordInputEmpty -> SignInResult.PasswordInputEmpty
+                    is SignInError.NotExistUsername -> SignInResult.NotExistUsername
+                    is SignInError.PasswordNotMatchingWithUsername -> SignInResult.PasswordNotMatchingWithUsername
+                    else -> null
+                }
+                _uiState.update { it.copy(isLoading = false, signInResult = errorResult) }
+                if (errorResult != null) {
+                    _effect.emit(SignInEffect.ShowSnackbar("Error"))
+                }
             }
         }
     }
 
-    fun trySignIn() {
+    private fun tryAutoSignIn() {
         viewModelScope.launch {
-            signInUseCase(usernameInput.value, passwordInput.value).onSuccess {
-                userRepository.saveAccount(Account(usernameInput.value, passwordInput.value))
-                signInUiState.emit(SignInUiState.Success)
+            _uiState.update { it.copy(isLoading = true) }
+            tryAutoSignInUseCase().onSuccess {
+                _uiState.update { it.copy(isLoading = false, signInResult = SignInResult.Success) }
             }.onFailure {
-                when (it) {
-                    is SignInError.UsernameInputEmpty -> signInUiState.emit(SignInUiState.UsernameInputEmpty)
-                    is SignInError.PasswordInputEmpty -> signInUiState.emit(SignInUiState.PasswordInputEmpty)
-                    is SignInError.NotExistUsername -> signInUiState.emit(SignInUiState.NotExistUsername)
-                    is SignInError.PasswordNotMatchingWithUsername -> signInUiState.emit(
-                        SignInUiState.PasswordNotMatchingWithUsername
-                    )
-                }
+                _uiState.update { it.copy(isLoading = false) }
             }
         }
     }
 }
 
-sealed interface SignInUiState {
-    data object Success : SignInUiState
-    data object UsernameInputEmpty : SignInUiState
-    data object PasswordInputEmpty : SignInUiState
-    data object NotExistUsername : SignInUiState
-    data object PasswordNotMatchingWithUsername : SignInUiState
+data class SignInUiState(
+    val username: String = "",
+    val password: String = "",
+    val isLoading: Boolean = false,
+    val signInResult: SignInResult? = null
+)
+
+sealed interface SignInIntent {
+    data class UpdateUsername(val username: String) : SignInIntent
+    data class UpdatePassword(val password: String) : SignInIntent
+    data object TrySignIn : SignInIntent
+    data object TryAutoSignIn : SignInIntent
+}
+
+sealed interface SignInResult {
+    data object Success : SignInResult
+    data object UsernameInputEmpty : SignInResult
+    data object PasswordInputEmpty : SignInResult
+    data object NotExistUsername : SignInResult
+    data object PasswordNotMatchingWithUsername : SignInResult
+}
+
+sealed interface SignInEffect {
+    data class ShowSnackbar(val message: String) : SignInEffect
```

Comment:
- 요렇게 바꾸면 좋을 것 같아요 ㅎㅎ SignUp 도요!
- 
```suggestion
    data class ShowSnackBar(val message: String) : SignInEffect
```